error: ISO C++17 does not allow incrementing expression of type bool [-Wincrement-bool]

Index: CPP/7zip/Archive/Wim/WimHandler.cpp
--- CPP/7zip/Archive/Wim/WimHandler.cpp.orig
+++ CPP/7zip/Archive/Wim/WimHandler.cpp
@@ -298,7 +298,7 @@ STDMETHODIMP CHandler::GetArchiveProperty(PROPID propI
 
       AString res;
 
-      bool numMethods = 0;
+      unsigned numMethods = 0;
       for (unsigned i = 0; i < ARRAY_SIZE(k_Methods); i++)
       {
         if (methodMask & ((UInt32)1 << i))
